docs/DeveloperGuide.adoc

```diff
@@ -222,42 +226,216 @@ image::CommitActivityDiagram.png[]
+* `Model#updateFilteredWorkoutList(Predicate<Workout> predicate)` - Updates the filter of `FilteredWorkoutList`
+
+To illustrate an example of a command from the Workout Manager, the following sequence diagram
+depicts flow of the program when the command `workout find w/Strength Training`
+
+image::WorkoutManagerSequenceDiagram.png[]
+
+Figure: Workout Manager Sequence Diagram
+
+. When the user runs the command `workout find w/Strength Training`, the `LogicManager` will first take in the command, by calling the `execute()` function on it.
+. Next, the `WorkoutCommandParser` has to `parseCommand()` for the command.
+. Once the command has been parsed as a `FindCommand`, it will be passed on to the `FindCommandParser`, which will then `findWorkout()` for the requested workout in the `Model`.
+. Finally, the requested `Workout` will be returned as the `CommandResult`.
+.. `Model#updateFilteredWorkoutList(Predicate<Workout> predicate)` is used to update the view of the workout list to show the requested Workout(s).
+
+==== Design Considerations
+
+===== Aspect: Exercises in workout
+
+* Option 1: Use existing Exercise class in workouts
+** Advantage: Building on an existing class is simpler and more intuitive
+** Disadvantage: Introduces a dependency on the Exercise class
+* Option 2: Create a new WorkoutExercise class for workouts
+** Advantage: Creates an extra layer of abstraction
+** Disadvantage: More code needed which may be redundant
+
+In the end, we decided to stick with Option 1. This is because creating a new WorkoutExercise class
+is redundant and unnecessary, when there is no functional difference between an Exercise and a WorkoutExercise,
+other than the context that they are referenced in. To simplify matters, using the existing Exercise class
+to construct workouts was better.
+// end::workout[]
+
+// tag::schedule[]
+=== Schedule
+==== Implementation
+The following sequence diagram shows how the `schedule create` operation works:
+
+image::schedule/ScheduleCreateSequenceDiagram.png[]
+
+
+The following sequence diagram shows how the `scheduledWorkout` are populated to user view:
+
+image::schedule/PopulateSortedScheduledWorkoutList.png[]
+
+==== Design Considerations
+
+===== Aspect: How to handle editing and deletion of recurring schedules
+
+* **Alternative 1:** Saves `scheduledWorkout` objects.
+** Pros: Easy to implement.
+** Cons: Hard to maintain when it comes to `edit` or `delete` of recurring scheduled workouts.
+* **Alternative 2 (current choice):** Saves `schedule` objects that can be used to produce `scheduledWorkout` when
+requested.
+** Pros: Changes to recurring schedules can populated to `scheduledWorkout` more easily.
+** Cons: Much more complicated model.
+// end::schedule[]
+
+// tag::log[]
+=== Log
+For logging, we propose saving the session data in the following class architecture.
+
+==== Implementation
+
+image::WorkoutSessionDiagram.png[]
```

Comment:
PlantUML by default uses square/circle/triangle to indicate visibility and symbols such as C, E for classes, enumerations etc. That is not standard UML, and needs to be changed. Please refer to PlantUML's documentation on how to do so.